Python/ceval.c

```diff
@@ -2124,6 +2124,19 @@ _PyEval_UnpackIterableStackRef(PyThreadState *tstate, _PyStackRef v_stackref,
         _PyErr_Format(tstate, PyExc_ValueError,
                       "too many values to unpack (expected %d)",
                       argcnt);
+        PyObject *exc = _PyErr_GetRaisedException(tstate);
```

Comment:
@encukou My recommendation is to introduce a new exception here that inherits from `ValueError` that has the extra required information because this feels a bit hacky but if you think that's too heavy I am happy to go this route but I want us to consider it.